readme.MD

```diff
@@ -1,16 +1,47 @@
 # Dialogs Framework
 The Dialogs Framework is a Python library for writing resumable functions with persistent execution state.
 
-# Examples
+## Resumable what now?
+This framework is intended for long-lived backend flows, such as chatbot conversations. It allows one to write code as a chronological flow, and provides the magic to make execution resume where it left off.
 
-## Set up
+## Code sample
+```python
+# No dialog, manual state manipulation
+def ask_a_name(client_answer, state):
+    if not state.did_ask_already:
+        state.did_ask_already = True
+        name = client_answer
+        return f"{name} is a beautiful name!"
+    else:
+        return "What is your name?"
```

Comment:
In a later section you can show how would it work without the framework and explain the drawbacks

Reply:
I want to keep them close together, I think seeing just the framework code isn't clear enough to the uninitiated.